### ISSUE TITLE
Referrals flag is missing from filters

### ISSUE THREAD

**Issue by PeaceGeeks** [PeaceGeeksSociety/services-advisor#89]
In the beta version, users can filter on services that require a filter. In this version, we would like to do the opposite - users should be able to search on services that do not require a referral.

**Comment by tuckerbuchy** [comment]
Is this basically just adding a filter that allows filtering by "referral" being present? Sorry not too familiar with beta version.

**Comment by amyngai** [comment]
I think beta version refers to this one: http://data.unhcr.org/jordan/services-advisor/
<img src="https://cloud.githubusercontent.com/assets/5739342/8236729/251a5af2-159f-11e5-9760-9a941f570646.png" width="400px">

So just to clarify @PeaceGeeks 

currently:
- Default search on everything
- Filter on if it needs a referral

change to:
- Default search on everything that needs a referral
- Filter on anything that _does not_ need a referral
- Checking the `do not need referral` filter would effectively get _more_ results. 

or:
- Default search on everything that needs a referral
- The `needs referral` button is always checked
- Users need to _uncheck_ the filter

is that correct?
Basically we can just change the `need referral` to `do not need referral` **OR** by default `need referral` is always checked _on_

If so, which is preferred?

**Comment by PeaceGeeks** [comment]
weird - why is it that I can sometimes flag @Edouard-Legoupil and sometimes cannot when I am replying on tix?

I would like his response on this, but my understanding is that he feels that: 
- default is to filter on all records
- users have the option to filter on only those services that do not require a referral
- also it would be great if the view of the filtered records (on the right) included this field

**Comment by amyngai** [comment]
(renee, it's likely because he's not a part of the organization so it doesn't show up automatically unless it's an issue that he created)

Ah! I understand. So one option (that we already have) `needs referral` and one added filter `does not need referral`

Is that correct?

**Comment by PeaceGeeks** [comment]
well i suppose i saw it like this: 
Option 1
- default - no referral filter 
- ability to filter on 'does not need referral'

But we could also do
Option 2
- default - no referral filter 
- ability to filter on 'does not need referral' AND on 'needs referral'

**Comment by PeaceGeeks** [comment]
@Edouard-Legoupil - please weigh in!

**Comment by PeaceGeeks** [comment]
@amyngai - thanks - i had actually created a UNHCR users which i had added him to, but I forgot to add that group to the repo. but its funny because I have been able to refer to him on other tix. anyway sorted for now.

**Comment by PeaceGeeks** [comment]
In absence of input from @Edouard-Legoupil who I know if dealing with personnel crisis right now, let's go with: 

Option 1
default - no referral filter
ability to filter on 'does not need referral'

**Comment by PeaceGeeks** [comment]
Is 'filter by org' supposed to appear in the area below the referral flag or is this something

**Comment by Edouard-Legoupil** [comment]
Sorry for my late response on this.

Actually one of the challenge of the app is that there are potentially two targeted audience:

For case worker, the goal is to provide the maximum searching capacity. Default should be all services and user should have the ability to filter on 'does not need referral' and on 'needs referral'

For refugees, the goal is to prevent misleading search. Default should be "no referral filter"  and user should not have the ability to filter on 'does need referral'

An idea could be to add this in the splash screen: https://github.com/PeaceGeeksSociety/services-advisor/issues/90

**Comment by tuckerbuchy** [comment]
What is the conclusion on this? Are we going with option 1?

**Comment by PeaceGeeks** [comment]
Yes I think that makes sense. So default would show all. Option to display/ filter on only those that do not require a referral (No Referral Required). In a next version with the splash screen, we might mix this up a bit, but this makes sense to me from a user perspective.

**Comment by lkysow** [comment]
@PeaceGeeks agreed.

@Edouard-Legoupil thanks for weighing in. I understand that there are different use cases however I think it's best to satisfy the use case of the refugees first since there will be lots of them using the app for the first time vs. case workers that will understand how it works as they continue to use it. Thus it makes more sense to me to optimize for the first-time users.

**Comment by tuckerbuchy** [comment]
PR:https://github.com/PeaceGeeksSociety/services-advisor/pull/104

**Comment by PeaceGeeks** [comment]
@Edouard-Legoupil - this suggestion does not make sense in terms of filtering because ALL services seem to require a referral. is that correct?

**Comment by Edouard-Legoupil** [comment]
Well it should not be the case -- see https://www.syrianrefugeeresponse.org/resources/sites/points?activity=5603 -- there's a few records with "Referral is not required"

**Comment by PeaceGeeks** [comment]
@Edouard-Legoupil tells me that there should be 6 "Referral is not required" records - also we should include in this filter "Referrals not accepted"

**Comment by PeaceGeeks** [comment]
@tuckerbuchy - as discussed the other night 'Referral Method" = "Referral is not required" AND "Referrals not accepted" is TRUE should be the filters for showing which services do not require a referral. Site is still returning 0 records no not closing yet!

**Comment by lkysow** [comment]
I've looked at the data and there are 6 different referral types + some services not having it defined at all. Given this I'm not sure how best to filter. @PeaceGeeks @Edouard-Legoupil 

"Email on a per case basis", "Referrals not accepted", "IA Form", "Telephone on a per case basis", "RAIS", "Referral is not required"

**Comment by PeaceGeeks** [comment]
@lkysow - if it is not a significant number, please just proceed with "Referrals not accepted", AND "Referral is not required" - Edouard can get the service providers to clean up the data. @Edouard-Legoupil - I have written to you on Skype about this, and will call you later to confirm. We will proceed with this unless we hear otherwise from you.

**Comment by PeaceGeeks** [comment]
@lkysow - it looks like you already did this (with the two methods above). I am going to close this for now. We can re-open if we hear from @Edouard-Legoupil

**Comment by PeaceGeeks** [comment]
re-opening - filter is returning 0 records - @lkysow, i thought you had fixed this, but not I'm not so sure? 

![screen shot 2015-07-23 at 10 04 25 pm](https://cloud.githubusercontent.com/assets/2263033/8865552/e5d178e0-3186-11e5-91c3-f8b11c3bd0bd.png)

**Comment by lkysow** [comment]
Okay so here's what I've done:
- any service marked "Referrals not accepted" or "Referral is not required" or not given a referral status is considered **Referral not required**
- any service marked otherwise, ex. "Email on a per case basis", "IA Form", "Telephone on a per case basis", "RAIS", is considered **Referral required**

I created 3 radio buttons: All, Referral Required, Referral Not Required. It turns out there's only 3 services where referrals are not required so it doesn't make sense to default to that, therefore I've defaulted to All
